clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java

```diff
@@ -59,17 +58,16 @@ public class LoginManager {
     private int refCount;
 
     private LoginManager(JaasContext jaasContext, String saslMechanism, Map<String, ?> configs,
-                 LoginMetadata<?> loginMetadata) throws LoginException {
+                 LoginMetadata<?> loginMetadata) throws Exception {
         this.loginMetadata = loginMetadata;
         this.login = Utils.newInstance(loginMetadata.loginClass);
         loginCallbackHandler = Utils.newInstance(loginMetadata.loginCallbackClass);
         try (AutoCloseable loginResources = this::closeResources) {
```

Comment:
Looking at this code again, not sure if this is correct. We want to call `closeResources` only if there is an exception, right? Won't this close the login even when login is successful?

Reply:
Sorry about the oversight. Correction made.